nixos/modules/services/games/vrising-server.nix

```diff
@@ -0,0 +1,174 @@
+              type = types.port;
+              default = 9877;
+              description = ''
+                Query port for the server, will be added to allowedUDPPorts.
+              '';
+            };
+          };
+        };
+        default = { };
+        example = literalExpression ''
+          {
+            Name = "V Rising Server";
+            Description = "";
+            MaxConnectedUsers = 5;
+            MaxConnectedAdmins = 5;
+            ServerFps = 30;
+            SaveName = "world1";
+            Secure = true;
+            ListOnMasterServer = false;
+            AutoSaveCount = 20;
+            AutoSaveInterval = 300;
+            GameSettingsPreset = "StandardPvE";
+            AdminOnlyDebugEvents = true;
+            DisableDebugEvents = false;
+          }'';
+        description = ''
+          Host settings as defined in <link xlink:href="https://github.com/StunlockStudios/vrising-dedicated-server-instructions">this repository</link>.
+          Look at the reference to see what you can configure in this attribute set.
+
+          Do not set <literal>Password</literal> here but use the dedicated <literal>passwordFile</literal> option instead.
+        '';
+      };
+
+      gameSettings = mkOption {
+        type = with types; attrsOf anything;
+        default = { };
+        example = literalExpression ''
+          {
+            GameModeType = "PvE";
+            DeathContainerPermission = "ClanMembers";
+            RelicSpawnType = "Plentiful";
+            CastleHeartDamageMode = "CanBeDestroyedOnlyWhenDecaying";
+            CastleDamageMode = "Never";
+            BloodBoundEquipment = true;
+            CanLootEnemyContainers = false;
+            Death_DurabilityFactorLoss = 0.0;
+            Death_DurabilityLossFactorAsResources = 0.0;
+            AnnounceSiegeWeaponSpawn = false;
+          }'';
+        description = ''
+          Game settings as defined in <link xlink:href="https://cdn.stunlock.com/blog/2022/05/25083113/Game-Server-Settings.pdf">this file</link>.
+          Look at the reference to see what you can configure in this attribute set.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    warnings = (if cfg.hostSettings ? Password then [''
+      You have defined a host password using the hostSettings option, which
+      will be ignored. Use the dedicated passwordFile option instead.
+    ''] else
+      [ ]);
+
+    networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [
+      cfg.hostSettings.port
+      cfg.hostSettings.queryPort
+    ];
+
+    systemd.services.vrising-server = {
+      description = "V Rising dedicated server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+
+      environment = {
+        # steamcmd uses $HOME/.local for temporary files
+        HOME = varLibStateDir;
+        saveDir = "./save-data";
+        hostSettings = builtins.toJSON cfg.hostSettings;
```

Comment:
that's not entirely what we meant. `pkgs.formats.json` has a `generate` that creates a formatted json file that can be used directly or piped through jq, no passing through env vars and running through `fromjson` necessary.

Reply:
I hope I got it right this time..